code/game/objects/items/mop.dm

```diff
@@ -18,11 +18,39 @@
 	var/mopspeed = 15
 	force_string = "robust... against germs"
 	var/insertable = TRUE
+	var/braced = FALSE
 
 /obj/item/mop/Initialize()
 	. = ..()
 	create_reagents(mopcap)
 
+/obj/item/mop/attack_self(mob/user)
+	if(user.mind.get_skill_level(/datum/skill/cleaning) >= SKILL_LEVEL_JOURNEYMAN)
+		if (!braced)
+			braced = TRUE
+			RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/moving_clean)
+			if(user.mind.get_skill_level(/datum/skill/cleaning) >= SKILL_EXP_MASTER)
+				to_chat(user, "<span class='notice'>You masterfully brace [src]! You are one with the mop, and can clean floors while running at full speed!</span>")
+			else
+				to_chat(user, "<span class='notice'>You skillfully brace [src]! You'll be able to clean floors while walking.</span>")
+		else
+			braced = FALSE
+			UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
+			to_chat(user, "<span class='notice'>You loosen your grip on [src].</span>")
+	else
+		to_chat(user, "<span class='warning'>You brace [src]... One day you'll be skilled enough to unlock its true potential, but not today...</span>")
+
+/obj/item/mop/proc/moving_clean(mob/user)
+	if(user.m_intent == MOVE_INTENT_WALK || user.mind.get_skill_level(/datum/skill/cleaning) >= SKILL_EXP_MASTER)
+		var/turf/target
+		target = user.loc
```

Comment:
user.loc isn't always a turf. I'm not sure if that makes a difference but the variable type is wrong.
It also doesn't always exist, again, no idea if that's important for the clean proc.
Should be var/atom/target

Reply:
would using `var/turf/target = get_turf(user)` be a better option?

Reply:
you want janitor cleaning from closed closet?